### ISSUE TITLE
Long-term preservation (for federally funded research code)

### ISSUE THREAD

**Issue by roomthily** [WhiteHouse/source-code-policy#128]
Related to #55.

Science/research code might be considered an edge case here, but still worth noting the preservation needs related to reproducibility/replicatability. In this case, not concerned with actively maintained (F)OSS code but code that's end-of-life in terms of active maintenance but that may have entered the scientific record through a research article (or other publication), software publication, or software citation or through inclusion in some workflow trace or other provenance output. 

Basically, how long will that software be available if no longer actively developed? Is it assigned a persistent unique identifier (resolvable) that can be managed such that a piece of EOL code is locatable? How and who is responsible for activities? 

We love shiny new code but there's still a significant crowd looking for and using 15 year old fortran out of the ARS so long-term preservation is important.

This also relates to discoverability, centralized registries and ongoing maintenance issues.
